src/Controller/ZoomController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Secret Santa project.
+ *
+ * (c) JoliCode <coucou@jolicode.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace JoliCode\SecretSanta\Controller;
+
+use JoliCode\SecretSanta\Application\ZoomApplication;
+use JoliCode\SecretSanta\Exception\AuthenticationException;
+use JoliCode\SecretSanta\Model\User;
+use League\OAuth2\Client\Provider\Zoom;
+use League\OAuth2\Client\Provider\ZoomResourceOwner;
+use League\OAuth2\Client\Token\AccessToken;
+use League\OAuth2\Client\Token\AccessTokenInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+
+class ZoomController extends AbstractController
+{
+    private $router;
+    private $zoomClientId;
+    private $zoomClientSecret;
+
+    public function __construct(string $zoomClientId, string $zoomClientSecret, RouterInterface $router)
+    {
+        $this->router = $router;
+        $this->zoomClientId = $zoomClientId;
+        $this->zoomClientSecret = $zoomClientSecret;
+    }
+
+    /**
+     * Ask for Zoom authentication and store the AccessToken in Session.
+     */
+    public function authenticate(Request $request, ZoomApplication $zoomApplication): Response
+    {
+        $session = $request->getSession();
+
+        $provider = new Zoom([
+            'clientId' => $this->zoomClientId,
+            'clientSecret' => $this->zoomClientSecret,
+            'redirectUri' => $this->router->generate('zoom_authenticate', [], RouterInterface::ABSOLUTE_URL),
+        ]);
+
+        if ($request->query->has('error')) {
```

Comment:
This bloc can be the first one of the method
(Note: many line can be re-organized to be closer to the place where they are used)